### PR TITLE
Add support for Security Group Adoption

### PR DESCRIPTION
**Pull request by pcolazurdo**
Issue #, if available: https://github.com/aws-controllers-k8s/community/issues/1946

Description of changes:

The existing logic doesn't allow for adoption of existing SecurityGroups because spec.Name is a required property for creation but the aws SDK returns GroupName. The deepCopy in sdkFind then is not enough because of the different names. 
By copying SecurityGroup GroupName into spec.Name as part of the sdk_read_many_post_set_output we solve this issue.

This fix allows adoption of existing security groups. Tests have been successful in at least two test scenarios: adopting an existing SG and then modifying its properties, trying to adopt an invalid SG.
I've also ran a few additional tests with the modified controller to be sure that existing functionality didn't break:
* Create a new SG
* Modify this SG
* Delete this SG

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: templates/hooks/security_group/sdk_read_many_post_set_output.go.tpl [hunk 1]
```diff
@@ -1,4 +1,8 @@
 	if found {
+
+	// Needed because SecurityGroups Name are held in GroupName property of the AWS resource
+        ko.Spec.Name = resp.SecurityGroups[0].GroupName
+
```
Comment (nnbu) [inline templates/hooks/security_group/sdk_read_many_post_set_output.go.tpl:5]: You also need to add generated file in this PR.

This seems to be a good approach to handle adoptedResource

**Review by nnbu — APPROVED** [review]

File: templates/hooks/security_group/sdk_read_many_post_set_output.go.tpl (hunk 1 above)
Comment (nnbu) [reply templates/hooks/security_group/sdk_read_many_post_set_output.go.tpl:5]: @pcolazurdo

Please let me know if you plan to make this change.

Comment (pcolazurdo) [reply templates/hooks/security_group/sdk_read_many_post_set_output.go.tpl:5]: Sorry, I was sick, new version with generated files should be in

**Comment by pcolazurdo** [comment]
@nnbu - I rebased against the latest release. Changes seem to be ready for merge

File: apis/v1alpha1/ack-generate-metadata.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ack_generate_info:
-  build_date: "2024-01-03T22:47:06Z"
-  build_hash: 00e081fb541587f33970ad80c99f2ac02e9c2525
+  build_date: "2024-01-08T17:38:00Z"
+  build_hash: 87c16d7c2c571fb6fc0e5299ccebbc9143ef64c4
   go_version: go1.21.5
-  version: v0.28.0-8-g00e081f
+  version: v0.27.1-19-g87c16d7
```
Comment (a-hilaly) [inline apis/v1alpha1/ack-generate-metadata.yaml:5]: Please regenerate using latest commit of code-generator

File: helm/values.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 image:
   repository: public.ecr.aws/aws-controllers-k8s/ec2-controller
-  tag: 1.2.0
+  tag: 1.1.1
```
Comment (a-hilaly) [inline helm/values.yaml:7]: Can you please pull the tags? we need to keep this 1.2.0 and later release a 1.2.1

Comment (pcolazurdo) [reply helm/values.yaml]: done

File: apis/v1alpha1/ack-generate-metadata.yaml (hunk 2 above)
Comment (pcolazurdo) [reply apis/v1alpha1/ack-generate-metadata.yaml]: done (thanks)

**Review by a-hilaly — APPROVED** [review]
Thank you @pcolazurdo 
/lgtm

**Pull request merged**